lib/mix/tasks/gh_pages_docs.ex

```diff
@@ -13,35 +11,32 @@ defmodule Mix.Tasks.GhPagesDocs do
   def run([version]) do
     version =
       case version do
-        "latest" -> "v#{Project.config()[:version]}"
+        "latest" -> prefix_tag(Mix.Project.config()[:version])
         tag -> prefix_tag(tag)
       end
 
+    # firstly we need to update version.js for the mix docs task
+    update_versions_js(version)
+
     Mix.Task.run("docs")
-    File.cd!("doc")
+
     0 = Mix.shell().cmd("git stash")
+    0 = Mix.shell().cmd("git checkout gh-pages")
 
-    # ignore all the changes, the only changes we want to track are in the
-    # doc directory which is listed in .gitignore
+    # secondly we do it again to avoid git conflicts from git stash pop
```

Comment:
I think I took a mental shortcut that was too big. What I meant is that we could use the stash entry with `versions.js` that was created before the checkout but it would cause a git conflict when executing `git stash pop`. I decided to invoke `update_versions_js/1` once again as a simpler way instead of resolving such conflict.
I'll try to make it more clear in this comment.